File: suchwaiting/src/main/java/com/azalea/suchwaiting/loadingimplemention/ILoadingRender.java
package com.azalea.suchwaiting.loadingimplemention;

import android.graphics.Canvas;
import android.graphics.ColorFilter;
import android.graphics.Rect;

import androidx.annotation.IntRange;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;


/**
 * interface of loading render
 */
public interface ILoadingRender {

	void draw(@NonNull Canvas canvas);

	void setAlpha(@IntRange(from = 0, to = 255) int alpha);

	void setColorFilter(@Nullable ColorFilter colorFilter);

	void onBoundsChange(Rect bounds);

	int getOpacity();

	int getIntrinsicHeight();

	int getIntrinsicWidth();
}

File: suchwaiting/src/main/java/com/azalea/suchwaiting/loadingimplemention/ILoadingRender.kt
package com.azalea.suchwaiting.loadingimplemention

import android.graphics.Canvas
import android.graphics.ColorFilter
import android.graphics.Rect
import androidx.annotation.IntRange

/**
 * interface of loading render
 */
interface ILoadingRender {
    fun draw(canvas: Canvas)
    fun setAlpha(@IntRange(from = 0, to = 255) alpha: Int)
    fun setColorFilter(colorFilter: ColorFilter?)
    fun onBoundsChange(bounds: Rect)
    fun setVisible(visible:Boolean)
    val opacity: Int
    val intrinsicHeight: Int
    val intrinsicWidth: Int
}